server/src/main/java/org/elasticsearch/search/query/QueryPhaseTimeoutException.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.query;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.search.SearchShardTarget;
+
+import java.io.IOException;
+
+/**
+ * Specific instance of QueryPhaseExecutionException that indicates that a search timeout occurred.
+ * Always returns http status 408 (request timeout)
+ */
+public class QueryPhaseTimeoutException extends QueryPhaseExecutionException {
```

Comment:
I'm afraid that introducing a new exception requires you to also list the exception under ElasticsearchException, so that the correct instance can be deserialized by the transport layer. This is the tricky bit: if we don't introduce new exceptions, we need to figure out how to customize the error code in SearchExceptions. If we do introduce new exceptions, we need to support them in the wire protocal.

Reply:
I registered it in the right places this time. I think in this case a new exception makes sense to disambiguate what status code should be returned.